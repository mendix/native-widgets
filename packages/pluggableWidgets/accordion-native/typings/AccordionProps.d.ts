/**
 * This file was generated from Accordion.xml
 * WARNING: All changes made to this file will be overwritten
 * @author Mendix UI Content Team
 */
import { ComponentType, CSSProperties, ReactNode } from "react";
import { ActionValue, DynamicValue, EditableValue, NativeIcon } from "mendix";

export type HeaderRenderModeEnum = "text" | "custom";

export type HeaderTextRenderModeEnum = "heading1" | "heading2" | "heading3" | "heading4" | "heading5" | "heading6";

export type GroupCollapsedEnum = "groupStartExpanded" | "groupStartCollapsed" | "groupStartDynamic";

export type AccessibleEnum = "yes" | "no";

export interface GroupsType {
    headerRenderMode: HeaderRenderModeEnum;
    headerTextRenderMode: HeaderTextRenderModeEnum;
    headerText: DynamicValue<string>;
    headerContent?: ReactNode;
    content?: ReactNode;
    visible: DynamicValue<boolean>;
    groupCollapsed: GroupCollapsedEnum;
    groupCollapsedDynamic?: DynamicValue<boolean>;
    accessible: AccessibleEnum;
    screenReaderCaption?: DynamicValue<string>;
    screenReaderHint?: DynamicValue<string>;
    groupCollapsedAttribute?: EditableValue<boolean>;
    groupOnChange?: ActionValue;
}

export type CollapseBehaviorEnum = "singleExpanded" | "multipleExpanded";

export type IconEnum = "right" | "left" | "no";

export interface GroupsPreviewType {
    headerRenderMode: HeaderRenderModeEnum;
    headerTextRenderMode: HeaderTextRenderModeEnum;
    headerText: string;
    headerContent: { widgetCount: number; renderer: ComponentType<{ caption?: string }> };
    content: { widgetCount: number; renderer: ComponentType<{ caption?: string }> };
    visible: string;
    groupCollapsed: GroupCollapsedEnum;
    groupCollapsedDynamic: string;
    accessible: AccessibleEnum;
    screenReaderCaption: string;
    screenReaderHint: string;
    groupCollapsedAttribute: string;
    groupOnChange: {} | null;
}

export interface AccordionProps<Style> {
    name: string;
    style: Style[];
    groups: GroupsType[];
    collapsible: boolean;
    collapseBehavior: CollapseBehaviorEnum;
    icon: IconEnum;
    iconCollapsed?: DynamicValue<NativeIcon>;
    iconExpanded?: DynamicValue<NativeIcon>;
}

export interface AccordionPreviewProps {
    className: string;
    style: string;
    styleObject?: CSSProperties;
    readOnly: boolean;
    groups: GroupsPreviewType[];
    collapsible: boolean;
    collapseBehavior: CollapseBehaviorEnum;
    icon: IconEnum;
    iconCollapsed: { type: "glyph"; iconClass: string; } | { type: "image"; imageUrl: string; } | null;
    iconExpanded: { type: "glyph"; iconClass: string; } | { type: "image"; imageUrl: string; } | null;
}
