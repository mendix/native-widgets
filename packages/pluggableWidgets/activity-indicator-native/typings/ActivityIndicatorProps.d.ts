/**
 * This file was generated from ActivityIndicator.xml
 * WARNING: All changes made to this file will be overwritten
 * @author Mendix UI Content Team
 */
import { CSSProperties } from "react";
import { DynamicValue } from "mendix";

export type AccessibleEnum = "yes" | "no";

export interface ActivityIndicatorProps<Style> {
    name: string;
    style: Style[];
    accessible: AccessibleEnum;
    screenReaderCaption?: DynamicValue<string>;
    screenReaderHint?: DynamicValue<string>;
}

export interface ActivityIndicatorPreviewProps {
    className: string;
    style: string;
    styleObject?: CSSProperties;
    readOnly: boolean;
    accessible: AccessibleEnum;
    screenReaderCaption: string;
    screenReaderHint: string;
}
