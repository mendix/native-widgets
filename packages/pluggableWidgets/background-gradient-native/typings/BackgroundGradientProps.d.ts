/**
 * This file was generated from BackgroundGradient.xml
 * WARNING: All changes made to this file will be overwritten
 * @author Mendix Widgets Framework Team
 */
import { ComponentType, CSSProperties, ReactNode } from "react";
import { ActionValue } from "mendix";
import { Big } from "big.js";

export interface ColorListType {
    color: string;
    offset: Big;
}

export interface ColorListPreviewType {
    color: string;
    offset: number | null;
}

export interface BackgroundGradientProps<Style> {
    name: string;
    style: Style[];
    colorList: ColorListType[];
    content: ReactNode;
    onClick?: ActionValue;
}

export interface BackgroundGradientPreviewProps {
    /**
     * @deprecated Deprecated since version 9.18.0. Please use class property instead.
     */
    className: string;
    class: string;
    style: string;
    styleObject?: CSSProperties;
    readOnly: boolean;
    colorList: ColorListPreviewType[];
    content: { widgetCount: number; renderer: ComponentType<{ children: ReactNode; caption?: string }> };
    onClick: {} | null;
}
