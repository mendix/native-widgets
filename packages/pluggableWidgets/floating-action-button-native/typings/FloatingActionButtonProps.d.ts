/**
 * This file was generated from FloatingActionButton.xml
 * WARNING: All changes made to this file will be overwritten
 * @author Mendix Widgets Framework Team
 */
import { CSSProperties } from "react";
import { ActionValue, DynamicValue, NativeIcon } from "mendix";

export type HorizontalPositionEnum = "left" | "center" | "right";

export type VerticalPositionEnum = "top" | "bottom";

export interface SecondaryButtonsType {
    icon: DynamicValue<NativeIcon>;
    caption?: DynamicValue<string>;
    onClick?: ActionValue;
}

export interface SecondaryButtonsPreviewType {
    icon: { type: "glyph"; iconClass: string; } | { type: "image"; imageUrl: string; iconUrl: string; } | { type: "icon"; iconClass: string; } | undefined;
    caption: string;
    onClick: {} | null;
}

export interface FloatingActionButtonProps<Style> {
    name: string;
    style: Style[];
    icon?: DynamicValue<NativeIcon>;
    iconActive?: DynamicValue<NativeIcon>;
    horizontalPosition: HorizontalPositionEnum;
    verticalPosition: VerticalPositionEnum;
    onClick?: ActionValue;
    secondaryButtons: SecondaryButtonsType[];
}

export interface FloatingActionButtonPreviewProps {
    /**
     * @deprecated Deprecated since version 9.18.0. Please use class property instead.
     */
    className: string;
    class: string;
    style: string;
    styleObject?: CSSProperties;
    readOnly: boolean;
    icon: { type: "glyph"; iconClass: string; } | { type: "image"; imageUrl: string; iconUrl: string; } | { type: "icon"; iconClass: string; } | undefined;
    iconActive: { type: "glyph"; iconClass: string; } | { type: "image"; imageUrl: string; iconUrl: string; } | { type: "icon"; iconClass: string; } | undefined;
    horizontalPosition: HorizontalPositionEnum;
    verticalPosition: VerticalPositionEnum;
    onClick: {} | null;
    secondaryButtons: SecondaryButtonsPreviewType[];
}
