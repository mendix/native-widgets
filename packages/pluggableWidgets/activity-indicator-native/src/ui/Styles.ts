import { Style } from "@mendix/piw-native-utils-internal";
import { ViewStyle } from "react-native";

export interface ActivityIndicatorStyle extends Style {
    container: ViewStyle;
    indicator: {
        size: "large" | "small";
        color: string;
    };
}

export const defaultActivityStyle: ActivityIndicatorStyle = {
    container: {},
    indicator: {
        size: "large",
        color: "gray"
    }
};
