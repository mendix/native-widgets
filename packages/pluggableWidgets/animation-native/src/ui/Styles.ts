import { Style } from "@mendix/piw-native-utils-internal";
import { ViewStyle } from "react-native";

export interface AnimationStyle extends Style {
    container: ViewStyle;
}

export const defaultAnimationStyle: AnimationStyle = {
    container: {
        // All ViewStyle properties are allowed
    }
};
