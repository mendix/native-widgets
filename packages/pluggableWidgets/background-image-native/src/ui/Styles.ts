import { ViewStyle, ImageStyle } from "react-native";
import { Style } from "@mendix/piw-native-utils-internal";

interface CustomImageStyle extends ImageStyle {
    svgColor?: string;
}

export interface BackgroundImageStyle extends Style {
    container: ViewStyle;
    image: CustomImageStyle;
}

export const defaultBackgroundImageStyle: BackgroundImageStyle = {
    container: {},
    image: {}
};
