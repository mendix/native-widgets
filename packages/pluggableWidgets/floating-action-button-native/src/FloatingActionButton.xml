<?xml version="1.0" encoding="utf-8" ?>
<widget id="com.mendix.widget.native.floatingactionbutton.FloatingActionButton" supportedPlatform="Native" needsEntityContext="false" offlineCapable="true" pluginWidget="true" xmlns="http://www.mendix.com/widget/1.0/" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://www.mendix.com/widget/1.0/ ../../../node_modules/mendix/custom_widget.xsd">
    <name>Floating action button</name>
    <description>Display a circular floating icon button.</description>
    <studioProCategory>Buttons</studioProCategory>
    <studioCategory>Buttons</studioCategory>
    <properties>
        <propertyGroup caption="General">
            <propertyGroup caption="Button">
                <property key="icon" type="icon" required="false">
                    <caption>Icon</caption>
                    <description />
                </property>
                <property key="iconActive" type="icon" required="false">
                    <caption>Secondary icon</caption>
                    <description>The secondary icon will be shown when the secondary buttons are visible.</description>
                </property>
            </propertyGroup>
            <propertyGroup caption="Position">
                <property key="horizontalPosition" type="enumeration" defaultValue="right">
                    <caption>Horizontal</caption>
                    <description />
                    <enumerationValues>
                        <enumerationValue key="left">Left</enumerationValue>
                        <enumerationValue key="center">Center</enumerationValue>
                        <enumerationValue key="right">Right</enumerationValue>
                    </enumerationValues>
                </property>
                <property key="verticalPosition" type="enumeration" defaultValue="bottom">
                    <caption>Vertical</caption>
                    <description />
                    <enumerationValues>
                        <enumerationValue key="top">Top</enumerationValue>
                        <enumerationValue key="bottom">Bottom</enumerationValue>
                    </enumerationValues>
                </property>
            </propertyGroup>
            <propertyGroup caption="Events">
                <property key="onClick" type="action">
                    <caption>On click</caption>
                    <description>This action will not be called when secondary buttons are used.</description>
                </property>
            </propertyGroup>
            <propertyGroup caption="Common">
                <systemProperty key="Name" />
            </propertyGroup>
            <propertyGroup caption="Accessibilty">
                <property key="accessible" type="enumeration" defaultValue="yes">
                    <caption>Accessible</caption>
                    <description />
                    <enumerationValues>
                        <enumerationValue key="yes">Yes</enumerationValue>
                        <enumerationValue key="no">No</enumerationValue>
                    </enumerationValues>
                </property>
                <property key="screenReaderCaption" type="textTemplate" required="false">
                    <caption>Screen reader caption</caption>
                    <description />
                </property>
            </propertyGroup>
        </propertyGroup>
        <propertyGroup caption="Secondary buttons">
            <property key="secondaryButtons" type="object" isList="true" required="false">
                <caption>Buttons</caption>
                <description>The secondary buttons animate in when the main button is pressed.</description>
                <properties>
                    <propertyGroup caption="Button">
                        <property key="icon" type="icon">
                            <caption>Icon</caption>
                            <description />
                        </property>
                        <property key="caption" type="textTemplate" required="false">
                            <caption>Caption</caption>
                            <description />
                        </property>
                        <property key="onClick" type="action">
                            <caption>On click</caption>
                            <description />
                        </property>
                        <property key="buttonAccessible" type="enumeration" defaultValue="yes">
                            <caption>Accessible</caption>
                            <description />
                            <enumerationValues>
                                <enumerationValue key="yes">Yes</enumerationValue>
                                <enumerationValue key="no">No</enumerationValue>
                            </enumerationValues>
                        </property>
                        <property key="buttonScreenReaderCaption" type="textTemplate" required="false">
                            <caption>Screen reader caption</caption>
                            <description />
                        </property>
                    </propertyGroup>
                </properties>
            </property>
        </propertyGroup>
    </properties>
</widget>
