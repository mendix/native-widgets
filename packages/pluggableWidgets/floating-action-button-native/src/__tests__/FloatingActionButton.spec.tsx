import { FloatingActionButtonProps, SecondaryButtonsType } from "../../typings/FloatingActionButtonProps";
import { FloatingActionButtonStyle } from "../ui/styles";
import { fireEvent, render, waitForElementToBeRemoved } from "@testing-library/react-native";
import { createElement } from "react";
import { FloatingActionButton } from "../FloatingActionButton";
import { actionValue, dynamicValue } from "@mendix/piw-utils-internal";
import { NativeIcon } from "mendix";
import { Icon } from "mendix/components/native/Icon";

describe("FloatingActionButton", () => {
    let defaultProps: FloatingActionButtonProps<FloatingActionButtonStyle>;
    const secondaryButtons: SecondaryButtonsType[] = [
        {
            icon: dynamicValue({
                type: "glyph",
                iconClass: "fa-glyph1"
            } as NativeIcon),
            caption: dynamicValue("caption1"),
            onClick: actionValue(true, false),
            buttonAccessible: "yes"
        },
        {
            icon: dynamicValue({
                type: "glyph",
                iconClass: "fa-glyph2"
            } as NativeIcon),
            caption: dynamicValue("caption2"),
            onClick: actionValue(true, false),
            buttonAccessible: "yes"
        },
        {
            icon: dynamicValue({
                type: "glyph",
                iconClass: "fa-glyph3"
            } as NativeIcon),
            caption: dynamicValue("caption3"),
            onClick: actionValue(true, false),
            buttonAccessible: "yes"
        }
    ];

    beforeEach(() => {
        defaultProps = {
            name: "FloatingAction",
            accessible: "yes",
            style: [],
            horizontalPosition: "right",
            verticalPosition: "bottom",
            secondaryButtons: []
        };
    });

    it("renders correct props without secondary buttons", () => {
        const component = render(<FloatingActionButton {...defaultProps} />);

        expect(component.toJSON()).toMatchSnapshot();
    });

    it("renders correct props with secondary buttons", () => {
        const component = render(<FloatingActionButton {...defaultProps} secondaryButtons={secondaryButtons} />);

        expect(component.toJSON()).toMatchSnapshot();
    });

    it("should open and close when clicked and secondary buttons are defined", async () => {
        const { getByTestId, queryAllByTestId } = render(
            <FloatingActionButton {...defaultProps} secondaryButtons={secondaryButtons} />
        );

        fireEvent(getByTestId("FloatingAction"), "onPress");
        expect(queryAllByTestId(/FloatingAction\$button*/)).toHaveLength(3);

        fireEvent(getByTestId("FloatingAction"), "onPress");
        await waitForElementToBeRemoved(() => queryAllByTestId("FloatingAction$button0"));
        expect(queryAllByTestId(/FloatingAction\$button*/)).toHaveLength(0);
    });

    it("should cancel any events of primary button if secondary buttons exist", () => {
        const mockEvent = actionValue(true, false);
        const { getByTestId } = render(
            <FloatingActionButton {...defaultProps} onClick={mockEvent} secondaryButtons={secondaryButtons} />
        );

        fireEvent(getByTestId("FloatingAction"), "onPress");
        expect(mockEvent.execute).not.toHaveBeenCalled();
    });

    it("should trigger event when clicked and no secondary buttons", () => {
        const mockEvent = actionValue(true, false);
        const { getByTestId } = render(<FloatingActionButton {...defaultProps} onClick={mockEvent} />);

        fireEvent(getByTestId("FloatingAction"), "onPress");
        expect(mockEvent.execute).toHaveBeenCalledTimes(1);
    });

    it("should trigger event on secondary button", () => {
        const { getByTestId } = render(<FloatingActionButton {...defaultProps} secondaryButtons={secondaryButtons} />);

        fireEvent(getByTestId("FloatingAction"), "onPress");
        fireEvent(getByTestId("FloatingAction$button1"), "onPress");
        expect(secondaryButtons[1].onClick?.execute).toHaveBeenCalledTimes(1);
    });

    it("should have custom icons on primary button", async () => {
        const icon = dynamicValue({
            type: "glyph",
            iconClass: "fa-glyphNormal"
        } as NativeIcon);
        const iconActive = dynamicValue({
            type: "glyph",
            iconClass: "fa-glyphActive"
        } as NativeIcon);
        const { getByTestId } = render(
            <FloatingActionButton
                {...defaultProps}
                icon={icon}
                iconActive={iconActive}
                secondaryButtons={secondaryButtons}
            />
        );
        const transformStyle = [{ transform: [{ rotate: "-180deg" }] }];

        const iconView = getByTestId("FloatingAction$IconView");
        const iconComponent = iconView.findByType(Icon);
        expect(iconComponent.props.icon).toEqual(icon.value);
        expect(iconView.props.style).not.toEqual(expect.arrayContaining(transformStyle));

        fireEvent(getByTestId("FloatingAction"), "onPress");

        const iconActiveComponent = iconView.findByType(Icon);
        expect(iconActiveComponent.props.icon).toEqual(iconActive.value);
        expect(iconView.props.style).toEqual(expect.arrayContaining(transformStyle));
    });

    it("should have custom icon on secondary button", async () => {
        const { getByTestId } = render(<FloatingActionButton {...defaultProps} secondaryButtons={secondaryButtons} />);

        fireEvent(getByTestId("FloatingAction"), "onPress");
        const secondaryButtonIcon = getByTestId("FloatingAction$button2").findByType(Icon);
        expect(secondaryButtonIcon.props.icon).toEqual(secondaryButtons[2].icon.value);
    });

    it("should have custom caption on secondary button", async () => {
        const { getByTestId, findByText } = render(
            <FloatingActionButton {...defaultProps} secondaryButtons={secondaryButtons} />
        );

        fireEvent(getByTestId("FloatingAction"), "onPress");
        await findByText(secondaryButtons[2].caption?.value as string);
    });

    describe("vertical position", () => {
        it("renders position top correctly", () => {
            defaultProps.verticalPosition = "top";
            const { toJSON } = render(<FloatingActionButton {...defaultProps} />);

            expect(toJSON()).toMatchSnapshot();
        });

        it("renders position bottom correctly", () => {
            const { toJSON } = render(<FloatingActionButton {...defaultProps} />);
            expect(toJSON()).toMatchSnapshot();
        });
    });
});
