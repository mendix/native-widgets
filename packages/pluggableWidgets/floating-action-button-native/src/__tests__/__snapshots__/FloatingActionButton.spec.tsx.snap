// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FloatingActionButton renders correct props with secondary buttons 1`] = `
<View
  pointerEvents="box-none"
  style={
    [
      [
        {
          "backgroundColor": "transparent",
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        },
        {
          "elevation": 5,
          "justifyContent": "flex-end",
          "zIndex": 999,
        },
      ],
      {
        "margin": 30,
      },
    ]
  }
>
  <View
    collapsable={false}
    pointerEvents="none"
    style={
      {
        "backgroundColor": "transparent",
        "bottom": 0,
        "elevation": 5,
        "justifyContent": "flex-end",
        "left": 0,
        "opacity": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
        "zIndex": 999,
      }
    }
  />
  <View
    pointerEvents="box-none"
    style={
      [
        [
          {
            "backgroundColor": "transparent",
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          },
          {
            "elevation": 5,
            "justifyContent": "flex-end",
            "zIndex": 999,
          },
        ],
        {
          "alignItems": "flex-end",
        },
        {
          "paddingVertical": 0,
        },
      ]
    }
  >
    <View
      style={
        [
          {
            "borderRadius": 27,
            "right": 0,
            "width": 54,
            "zIndex": 999,
          },
          {
            "elevation": 5,
            "shadowColor": "#000",
            "shadowOffset": {
              "height": 5,
              "width": 0,
            },
            "shadowOpacity": 0.35,
            "shadowRadius": 3,
          },
          {
            "backgroundColor": "#e74c3c",
            "elevation": 5,
            "shadowColor": "#000",
            "shadowOffset": {
              "height": 5,
              "width": 0,
            },
            "shadowOpacity": 0.3,
            "shadowRadius": 3,
            "size": 54,
          },
        ]
      }
    >
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        nativeBackgroundAndroid={
          {
            "attribute": "selectableItemBackground",
            "rippleRadius": undefined,
            "type": "ThemeAttrAndroid",
          }
        }
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "backgroundColor": "rgba(0, 0, 0, 0)",
            "borderRadius": 27,
            "height": 54,
            "width": 54,
          }
        }
        testID="FloatingAction"
      >
        <View
          collapsable={false}
          style={
            {
              "alignItems": "center",
              "borderRadius": 27,
              "height": 54,
              "justifyContent": "center",
              "transform": [
                {
                  "scale": 1,
                },
                {
                  "rotate": "0deg",
                },
              ],
              "width": 54,
            }
          }
        >
          <View
            style={
              [
                {},
                {
                  "alignItems": "center",
                  "flexShrink": 1,
                  "justifyContent": "center",
                },
              ]
            }
            testID="FloatingAction$IconView"
          >
            <View
              testId="icon"
            />
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;

exports[`FloatingActionButton renders correct props without secondary buttons 1`] = `
<View
  pointerEvents="box-none"
  style={
    [
      [
        {
          "backgroundColor": "transparent",
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        },
        {
          "elevation": 5,
          "justifyContent": "flex-end",
          "zIndex": 999,
        },
      ],
      {
        "margin": 30,
      },
    ]
  }
>
  <View
    collapsable={false}
    pointerEvents="none"
    style={
      {
        "backgroundColor": "transparent",
        "bottom": 0,
        "elevation": 5,
        "justifyContent": "flex-end",
        "left": 0,
        "opacity": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
        "zIndex": 999,
      }
    }
  />
  <View
    pointerEvents="box-none"
    style={
      [
        [
          {
            "backgroundColor": "transparent",
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          },
          {
            "elevation": 5,
            "justifyContent": "flex-end",
            "zIndex": 999,
          },
        ],
        {
          "alignItems": "flex-end",
        },
        {
          "paddingVertical": 0,
        },
      ]
    }
  >
    <View
      style={
        [
          {
            "borderRadius": 27,
            "right": 0,
            "width": 54,
            "zIndex": 999,
          },
          {
            "elevation": 5,
            "shadowColor": "#000",
            "shadowOffset": {
              "height": 5,
              "width": 0,
            },
            "shadowOpacity": 0.35,
            "shadowRadius": 3,
          },
          {
            "backgroundColor": "#e74c3c",
            "elevation": 5,
            "shadowColor": "#000",
            "shadowOffset": {
              "height": 5,
              "width": 0,
            },
            "shadowOpacity": 0.3,
            "shadowRadius": 3,
            "size": 54,
          },
        ]
      }
    >
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        nativeBackgroundAndroid={
          {
            "attribute": "selectableItemBackground",
            "rippleRadius": undefined,
            "type": "ThemeAttrAndroid",
          }
        }
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "backgroundColor": "rgba(0, 0, 0, 0)",
            "borderRadius": 27,
            "height": 54,
            "width": 54,
          }
        }
        testID="FloatingAction"
      >
        <View
          collapsable={false}
          style={
            {
              "alignItems": "center",
              "borderRadius": 27,
              "height": 54,
              "justifyContent": "center",
              "transform": [
                {
                  "scale": 1,
                },
                {
                  "rotate": "0deg",
                },
              ],
              "width": 54,
            }
          }
        >
          <View
            style={
              [
                {},
                {
                  "alignItems": "center",
                  "flexShrink": 1,
                  "justifyContent": "center",
                },
              ]
            }
            testID="FloatingAction$IconView"
          >
            <View
              testId="icon"
            />
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;

exports[`FloatingActionButton vertical position renders position bottom correctly 1`] = `
<View
  pointerEvents="box-none"
  style={
    [
      [
        {
          "backgroundColor": "transparent",
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        },
        {
          "elevation": 5,
          "justifyContent": "flex-end",
          "zIndex": 999,
        },
      ],
      {
        "margin": 30,
      },
    ]
  }
>
  <View
    collapsable={false}
    pointerEvents="none"
    style={
      {
        "backgroundColor": "transparent",
        "bottom": 0,
        "elevation": 5,
        "justifyContent": "flex-end",
        "left": 0,
        "opacity": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
        "zIndex": 999,
      }
    }
  />
  <View
    pointerEvents="box-none"
    style={
      [
        [
          {
            "backgroundColor": "transparent",
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          },
          {
            "elevation": 5,
            "justifyContent": "flex-end",
            "zIndex": 999,
          },
        ],
        {
          "alignItems": "flex-end",
        },
        {
          "paddingVertical": 0,
        },
      ]
    }
  >
    <View
      style={
        [
          {
            "borderRadius": 27,
            "right": 0,
            "width": 54,
            "zIndex": 999,
          },
          {
            "elevation": 5,
            "shadowColor": "#000",
            "shadowOffset": {
              "height": 5,
              "width": 0,
            },
            "shadowOpacity": 0.35,
            "shadowRadius": 3,
          },
          {
            "backgroundColor": "#e74c3c",
            "elevation": 5,
            "shadowColor": "#000",
            "shadowOffset": {
              "height": 5,
              "width": 0,
            },
            "shadowOpacity": 0.3,
            "shadowRadius": 3,
            "size": 54,
          },
        ]
      }
    >
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        nativeBackgroundAndroid={
          {
            "attribute": "selectableItemBackground",
            "rippleRadius": undefined,
            "type": "ThemeAttrAndroid",
          }
        }
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "backgroundColor": "rgba(0, 0, 0, 0)",
            "borderRadius": 27,
            "height": 54,
            "width": 54,
          }
        }
        testID="FloatingAction"
      >
        <View
          collapsable={false}
          style={
            {
              "alignItems": "center",
              "borderRadius": 27,
              "height": 54,
              "justifyContent": "center",
              "transform": [
                {
                  "scale": 1,
                },
                {
                  "rotate": "0deg",
                },
              ],
              "width": 54,
            }
          }
        >
          <View
            style={
              [
                {},
                {
                  "alignItems": "center",
                  "flexShrink": 1,
                  "justifyContent": "center",
                },
              ]
            }
            testID="FloatingAction$IconView"
          >
            <View
              testId="icon"
            />
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;

exports[`FloatingActionButton vertical position renders position top correctly 1`] = `
<View
  pointerEvents="box-none"
  style={
    [
      [
        {
          "backgroundColor": "transparent",
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        },
        {
          "elevation": 5,
          "justifyContent": "flex-start",
          "zIndex": 999,
        },
      ],
      {
        "margin": 30,
      },
    ]
  }
>
  <View
    collapsable={false}
    pointerEvents="none"
    style={
      {
        "backgroundColor": "transparent",
        "bottom": 0,
        "elevation": 5,
        "justifyContent": "flex-start",
        "left": 0,
        "opacity": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
        "zIndex": 999,
      }
    }
  />
  <View
    pointerEvents="box-none"
    style={
      [
        [
          {
            "backgroundColor": "transparent",
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          },
          {
            "elevation": 5,
            "justifyContent": "flex-start",
            "zIndex": 999,
          },
        ],
        {
          "alignItems": "flex-end",
        },
        {
          "paddingVertical": 0,
        },
      ]
    }
  >
    <View
      style={
        [
          {
            "borderRadius": 27,
            "right": 0,
            "width": 54,
            "zIndex": 999,
          },
          {
            "elevation": 5,
            "shadowColor": "#000",
            "shadowOffset": {
              "height": 5,
              "width": 0,
            },
            "shadowOpacity": 0.35,
            "shadowRadius": 3,
          },
          {
            "backgroundColor": "#e74c3c",
            "elevation": 5,
            "shadowColor": "#000",
            "shadowOffset": {
              "height": 5,
              "width": 0,
            },
            "shadowOpacity": 0.3,
            "shadowRadius": 3,
            "size": 54,
          },
        ]
      }
    >
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        nativeBackgroundAndroid={
          {
            "attribute": "selectableItemBackground",
            "rippleRadius": undefined,
            "type": "ThemeAttrAndroid",
          }
        }
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "backgroundColor": "rgba(0, 0, 0, 0)",
            "borderRadius": 27,
            "height": 54,
            "width": 54,
          }
        }
        testID="FloatingAction"
      >
        <View
          collapsable={false}
          style={
            {
              "alignItems": "center",
              "borderRadius": 27,
              "height": 54,
              "justifyContent": "center",
              "transform": [
                {
                  "scale": 1,
                },
                {
                  "rotate": "0deg",
                },
              ],
              "width": 54,
            }
          }
        >
          <View
            style={
              [
                {},
                {
                  "alignItems": "center",
                  "flexShrink": 1,
                  "justifyContent": "center",
                },
              ]
            }
            testID="FloatingAction$IconView"
          >
            <View
              testId="icon"
            />
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;
